File: MainWindowTabHardwareInfo.vb
Imports Subtitle_draft_GMTPC.Services

Partial Class MainWindow

#Region "Hardware Info - Fields"

    Private _isHardwareLoading As Boolean = False

#End Region

#Region "Hardware Info - Methods"

    ''' <summary>
    ''' Load toàn bộ thông tin phần cứng khi mở app
    ''' </summary>
    Private Sub LoadHardwareInfo()
        If _isHardwareLoading Then Return
        Try
            _isHardwareLoading = True

            TxtGpuInfo.Text = HardwareInfoService.GetGpuInfo()
            TxtCpuInfo.Text = HardwareInfoService.GetCpuInfo()
            TxtRamInfo.Text = HardwareInfoService.GetRamInfo()
            TxtMainboardInfo.Text = HardwareInfoService.GetMainboardInfo()
        Catch ex As Exception
            TxtGpuInfo.Text = String.Format("Lỗi: {0}", ex.Message)
            TxtCpuInfo.Text = String.Format("Lỗi: {0}", ex.Message)
            TxtRamInfo.Text = String.Format("Lỗi: {0}", ex.Message)
            TxtMainboardInfo.Text = String.Format("Lỗi: {0}", ex.Message)
        Finally
            _isHardwareLoading = False
        End Try
    End Sub

#End Region

#Region "Hardware Info - Toast"

    Private Async Sub ShowToastHardware(message As String)
        ToastTextHardware.Text = message
        ToastBorderHardware.Visibility = Visibility.Visible
        Await Task.Delay(2000)
        ToastBorderHardware.Visibility = Visibility.Collapsed
    End Sub

#End Region

#Region "Hardware Info - Copy Buttons"

    Private Sub BtnCopyGpu_Click(sender As Object, e As RoutedEventArgs)
        If String.IsNullOrWhiteSpace(TxtGpuInfo.Text) Then Return
        Try
            Clipboard.SetText(TxtGpuInfo.Text)
            ShowToastHardware("📋 Đã copy GPU info!")
        Catch ex As Exception
            MessageBox.Show("Lỗi: " & ex.Message, "Lỗi", MessageBoxButton.OK, MessageBoxImage.Error)
        End Try
    End Sub

    Private Sub BtnCopyCpu_Click(sender As Object, e As RoutedEventArgs)
        If String.IsNullOrWhiteSpace(TxtCpuInfo.Text) Then Return
        Try
            Clipboard.SetText(TxtCpuInfo.Text)
            ShowToastHardware("📋 Đã copy CPU info!")
        Catch ex As Exception
            MessageBox.Show("Lỗi: " & ex.Message, "Lỗi", MessageBoxButton.OK, MessageBoxImage.Error)
        End Try
    End Sub

    Private Sub BtnCopyRam_Click(sender As Object, e As RoutedEventArgs)
        If String.IsNullOrWhiteSpace(TxtRamInfo.Text) Then Return
        Try
            Clipboard.SetText(TxtRamInfo.Text)
            ShowToastHardware("📋 Đã copy RAM info!")
        Catch ex As Exception
            MessageBox.Show("Lỗi: " & ex.Message, "Lỗi", MessageBoxButton.OK, MessageBoxImage.Error)
        End Try
    End Sub

    Private Sub BtnCopyMainboard_Click(sender As Object, e As RoutedEventArgs)
        If String.IsNullOrWhiteSpace(TxtMainboardInfo.Text) Then Return
        Try
            Clipboard.SetText(TxtMainboardInfo.Text)
            ShowToastHardware("📋 Đã copy Mainboard info!")
        Catch ex As Exception
            MessageBox.Show("Lỗi: " & ex.Message, "Lỗi", MessageBoxButton.OK, MessageBoxImage.Error)
        End Try
    End Sub

#End Region

End Class

File: MainWindowTabHardwareInfo.cs
using System;
using System.Threading.Tasks;
using System.Windows;
using Subtitle_draft_GMTPC.Services;

namespace Subtitle_draft_GMTPC
{
    public partial class MainWindow : Window
    {
        #region Hardware Info - Fields

        private bool _isHardwareLoading = false;

        #endregion

        #region Hardware Info - Methods

        /// <summary>
        /// Load toàn bộ thông tin phần cứng khi mở app.
        /// </summary>
        private void LoadHardwareInfo()
        {
            if (_isHardwareLoading) return;

            try
            {
                _isHardwareLoading = true;

                TxtGpuInfo.Text = HardwareInfoService.GetGpuInfo();
                TxtCpuInfo.Text = HardwareInfoService.GetCpuInfo();
                TxtRamInfo.Text = HardwareInfoService.GetRamInfo();
                TxtMainboardInfo.Text = HardwareInfoService.GetMainboardInfo();
            }
            catch (Exception ex)
            {
                TxtGpuInfo.Text = $"Lỗi: {ex.Message}";
                TxtCpuInfo.Text = $"Lỗi: {ex.Message}";
                TxtRamInfo.Text = $"Lỗi: {ex.Message}";
                TxtMainboardInfo.Text = $"Lỗi: {ex.Message}";
            }
            finally
            {
                _isHardwareLoading = false;
            }
        }

        #endregion

        #region Hardware Info - Toast

        private async void ShowToastHardware(string message)
        {
            ToastTextHardware.Text = message;
            ToastBorderHardware.Visibility = Visibility.Visible;
            await Task.Delay(2000);
            ToastBorderHardware.Visibility = Visibility.Collapsed;
        }

        #endregion

        #region Hardware Info - Copy Buttons

        private void BtnCopyGpu_Click(object sender, RoutedEventArgs e)
        {
            if (string.IsNullOrWhiteSpace(TxtGpuInfo.Text)) return;

            try
            {
                Clipboard.SetText(TxtGpuInfo.Text);
                ShowToastHardware("📋 Đã copy thông tin GPU.");
            }
            catch (Exception ex)
            {
                MessageBox.Show("Lỗi: " + ex.Message, "Lỗi", MessageBoxButton.OK, MessageBoxImage.Error);
            }
        }

        private void BtnCopyCpu_Click(object sender, RoutedEventArgs e)
        {
            if (string.IsNullOrWhiteSpace(TxtCpuInfo.Text)) return;

            try
            {
                Clipboard.SetText(TxtCpuInfo.Text);
                ShowToastHardware("📋 Đã copy thông tin CPU.");
            }
            catch (Exception ex)
            {
                MessageBox.Show("Lỗi: " + ex.Message, "Lỗi", MessageBoxButton.OK, MessageBoxImage.Error);
            }
        }

        private void BtnCopyRam_Click(object sender, RoutedEventArgs e)
        {
            if (string.IsNullOrWhiteSpace(TxtRamInfo.Text)) return;

            try
            {
                Clipboard.SetText(TxtRamInfo.Text);
                ShowToastHardware("📋 Đã copy thông tin RAM.");
            }
            catch (Exception ex)
            {
                MessageBox.Show("Lỗi: " + ex.Message, "Lỗi", MessageBoxButton.OK, MessageBoxImage.Error);
            }
        }

        private void BtnCopyMainboard_Click(object sender, RoutedEventArgs e)
        {
            if (string.IsNullOrWhiteSpace(TxtMainboardInfo.Text)) return;

            try
            {
                Clipboard.SetText(TxtMainboardInfo.Text);
                ShowToastHardware("📋 Đã copy thông tin Mainboard.");
            }
            catch (Exception ex)
            {
                MessageBox.Show("Lỗi: " + ex.Message, "Lỗi", MessageBoxButton.OK, MessageBoxImage.Error);
            }
        }

        #endregion
    }
}
